eos-tech-support/eos-list-users

```diff
@@ -1,27 +1,41 @@
 #!/bin/bash -e
 
-# Lists all users (i.e., all home directories on /dev/sda2)
-
-DEVICE=/dev/sda2
-MOUNT=/mnt/sda2
+# Lists all users (i.e., all home directories on all ostree devices)
 
 USERID=$(id -u)
 if [ "$USERID" != "0" ]; then
     echo "Program requires superuser privileges"
     exit 1
 fi
 
+MOUNT=`mktemp -d`
+
 cleanup() {
     set +e
-    umount $MOUNT
+    if mount | grep $MOUNT; then
+        umount $MOUNT
+    fi
     rmdir $MOUNT
 }
 trap cleanup EXIT
 
 mkdir -p $MOUNT
-mount $DEVICE $MOUNT
 
-# FIXME A more robust approach would be the following:
-# - read /usr/etc/login.defs and extract UID_MIN
-# - read /etc/passwd and get all users with uid >= UID_MIN
-ls $MOUNT/home
+DEVICES=`blkid -t LABEL=ostree | awk -F ":" '{print $1}'`
```

Comment:
I tested by running from USB on an EC-100 eMMC running an ostree image, and did not see any /dev/block/* devices.  I also just tested locally on my dev-converted laptop and I still don't see any /dev/block/* device:

```
roddy@endless:~/checkout/eos-meta/eos-tech-support$ sudo ./eos-list-users 
[sudo] password for roddy: 

Users on /dev/sda3:
roddy  shared

roddy@endless:~/checkout/eos-meta/eos-tech-support$ 
```